src/celimage/bmp.cpp

```diff
@@ -26,120 +27,100 @@ namespace
 typedef struct
 {
     unsigned char magic[2];
-    uint32_t size;
-    uint32_t reserved;
-    uint32_t offset;
+    std::uint32_t size;
+    std::uint32_t reserved;
+    std::uint32_t offset;
 } BMPFileHeader;
 
 typedef struct
 {
-    uint32_t size;
-    int32_t width;
-    int32_t height;
-    uint16_t planes;
-    uint16_t bpp;
-    uint32_t compression;
-    uint32_t imageSize;
-    int32_t widthPPM;
-    int32_t heightPPM;
-    uint32_t colorsUsed;
-    uint32_t colorsImportant;
+    std::uint32_t size;
+    std::int32_t width;
+    std::int32_t height;
+    std::uint16_t planes;
+    std::uint16_t bpp;
+    std::uint32_t compression;
+    std::uint32_t imageSize;
+    std::int32_t widthPPM;
+    std::int32_t heightPPM;
+    std::uint32_t colorsUsed;
+    std::uint32_t colorsImportant;
 } BMPImageHeader;
 
 
-int32_t readInt32(ifstream& in)
-{
-    uint8_t b[4];
-    in.read(reinterpret_cast<char*>(b), 4);
-    int32_t val = ((int32_t) b[3] << 24) +
-                  ((int32_t) b[2] << 16) +
-                  ((int32_t) b[1] << 8) +
-                  ((int32_t) b[0]);
-    LE_TO_CPU_INT32(val, val);
-    return val;
-}
-
-int16_t readInt16(ifstream& in)
-{
-    uint8_t b[2];
-    in.read(reinterpret_cast<char*>(b), 2);
-    int16_t val = ((int16_t) b[1] << 8) + (int16_t) b[0];
-    LE_TO_CPU_INT16(val, val);
-    return val;
-}
-
-
-Image* LoadBMPImage(ifstream& in)
+Image* LoadBMPImage(std::istream& in)
 {
     BMPFileHeader fileHeader;
     BMPImageHeader imageHeader;
-    uint8_t* pixels;
-
-    in >> fileHeader.magic[0];
-    in >> fileHeader.magic[1];
-    fileHeader.size = readInt32(in);
-    fileHeader.reserved = readInt32(in);
-    fileHeader.offset = readInt32(in);
-
-    if (fileHeader.magic[0] != 'B' || fileHeader.magic[1] != 'M')
-        return nullptr;
 
-    imageHeader.size = readInt32(in);
-    imageHeader.width = readInt32(in);
-    imageHeader.height = readInt32(in);
-    imageHeader.planes = readInt16(in);
-    imageHeader.bpp = readInt16(in);
-    imageHeader.compression = readInt32(in);
-    imageHeader.imageSize = readInt32(in);
-    imageHeader.widthPPM = readInt32(in);
-    imageHeader.heightPPM = readInt32(in);
-    imageHeader.colorsUsed = readInt32(in);
-    imageHeader.colorsImportant = readInt32(in);
-
-    if (imageHeader.width <= 0 || imageHeader.height <= 0)
-        return nullptr;
-
-    // We currently don't support compressed BMPs
-    if (imageHeader.compression != 0)
-        return nullptr;
-    // We don't handle 1-, 2-, or 4-bpp images
-    if (imageHeader.bpp != 8 && imageHeader.bpp != 24 && imageHeader.bpp != 32)
+    if (!celutil::read_le(in, fileHeader.magic[0])
```

Comment:
there are places where you use read_le with explicitly defined type and here without it. also according to the current style it should be named ReadLE

Reply:
Thanks, fixed.